src/components/core/Bookmark/index.js

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useEffect } from "react";
+import { useQuery } from "react-query";
+import { useAuth } from "../../../contexts/auth";
+import { useRouter } from "next/router";
+import { useUpdateBookmarks } from "hooks";
+import BookmarkIcon from "./BookmarkIcon";
+import LoginModal from "./loginmodal";
+import api from "api";
+
+const Bookmark = () => {
+  const [isBookmarked, setBookmark] = useState("");
+  const { authorize, isAuthorized, user } = useAuth();
+  const [isModalOpen, setModalOpen] = useState(false);
```

Comment:
```suggestion
  const [isModalOpen, setIsModalOpen] = useState(false);
```
Keep the naming convention of `useState` hook.